From ab6262344fdec52cd4087ddcccf9911537c1e5b3 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 12:26:02 -0400
Subject: [ethernet] tg3: remove empty MDIO bus reset function

Message-id: <1410179173-14304-9-git-send-email-ivecera@redhat.com>
Patchwork-id: 91669
O-Subject: [RHEL7.1 PATCH 08/19] tg3: remove empty MDIO bus reset function
Bugzilla: 1088637
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: #1088637

Upstream commit(s):
commit 7427440b2a7158d047f93bcf2c39472eae88acba
Author: Florian Fainelli <f.fainelli@gmail.com>
Date:   Wed Mar 26 18:07:14 2014 -0700

    tg3: remove empty MDIO bus reset function

    tg3_mdio_reset() does nothing useful and is optional for the MDIO bus
    code, so let's just remove it.

    Signed-off-by: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/tg3.c b/drivers/net/ethernet/broadcom/tg3.c
index 9e535c7..150580d 100644
--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -1401,11 +1401,6 @@ static int tg3_mdio_write(struct mii_bus *bp, int mii_id, int reg, u16 val)
  return ret;
 }
 
-static int tg3_mdio_reset(struct mii_bus *bp)
-{
- return 0;
-}
-
 static void tg3_mdio_config_5785(struct tg3 *tp)
 {
  u32 val;
@@ -1542,7 +1537,6 @@ static int tg3_mdio_init(struct tg3 *tp)
  tp->mdio_bus->parent   = &tp->pdev->dev;
  tp->mdio_bus->read     = &tg3_mdio_read;
  tp->mdio_bus->write    = &tg3_mdio_write;
- tp->mdio_bus->reset    = &tg3_mdio_reset;
  tp->mdio_bus->phy_mask = ~(1 << tp->phy_addr);
  tp->mdio_bus->irq      = &tp->mdio_irq[0];
 
-- 
1.7.1